### ISSUE TITLE
Freezing air too expensive

### ISSUE THREAD

**Issue by NeroOneTrueKing** [GTNewHorizons/NewHorizons#3128]
#### Which modpack version are you using?
2.0.5
#
#### If in multiplayer; On which server does this happen? 
Private LAN
#
#### What did you try to do, and what did you expect to happen?
Freeze thousands of air cells (with 4x IV freezers) to make liquid air to get argon + other gases
#
#### What happend instead? (Attach screenshots if needed)
It took much, much longer than expected.
Right now, freezing a single air cell is ~1million eu at 480 eu/t, over 120 seconds:
<img width="266" alt="liquid air" src="https://user-images.githubusercontent.com/28197216/41478563-d0432d04-708d-11e8-91d3-579ee861e5e8.PNG">

#
#### What do you suggest instead/what changes do you propose?
I think the time per cell should be drastically reduced to 12 seconds per cell or maybe less. There have been multiple issues (#2982, #3057) and commits due to too-long vacuum freezer recipes, so it is reasonable for this to be changed as well.

**Comment by Dream-Master** [comment]
ok maybe 2 mins to expensive but 12 secs only to short. cooling air is more energy costly than 12 secs.

**Comment by Dream-Master** [comment]
@Anarack explain your math a bit. i dont understand it.

**Comment by Anarack** [comment]
ok previous napkin math was wrong on the multiblock count, somehow got double what you need, but here you go.

1 liquid air centrifuge need 53 liquid air cells per operation to run at 74.2 seconds per operation.
1 vacuum freezer at 12 seconds per operation produces 1 cell every 12 seconds.
So you need, 53 cells * 12 seconds, 636 seconds of vacuum freezer time per liquid air operation.
To do this you need to run 9 vacume freezers at HV to sustain 1 liquid air centrifuge running at full time.

Argon production per centrifuge.
1000L / 74.2 seconds = 13.4 L/s argon

so you would be paying 4,325 eu/t before loss for 13.4 L/s of argon.

or the gt++ rout you would use 1 and 1/8  factory grade freezers but there is no recipe for compress air to liquid air in NEI at this moment. It would use less power, by about half + the power requirements for producing cryothium.

**Comment by Anarack** [comment]
For further reference, this is cost from UU matter.
(this is not including the gt++ matter fabricator, idk what bonuses it has)
1KL argon needs 40L UU + 614,400 EU
40L UU needs 8,222,720 EU and 40L UU AMP
40L UU AMP needs 360 scrap and 230,400 EU
360 scrap (Assuming we are consistent with ic2 scrap ratios) needs 16,200 EU
(not counting loss or cost to produce junk items)

total from UU matter argon is 9,083 EU/L of argon
the above method at 12 seconds freeze time is 6,455 EU/L argon.
IDK if it should be more or less expensive than UU, but I lean to less just for more unique set ups

**Comment by Kortako** [comment]
Also FYI, from looking at [this change](https://github.com/GTNewHorizons/GT5-Unofficial/commit/5cb0bfb2543424d1d0a8feb12d47dd8ec1dc8a2f#diff-9d48c6c95ad1120b9a2be1e2ff89f445L1473), I believe the previous time for that specific recipe was 25 sec at MV(?), so it seems like the time got nerfed quite a bit.

**Comment by 0lafe** [comment]
With the old times, it was still a pain. The problem is with vac freezers you can't speed up recipes in any way other than making more of them, or overclocking. Overclocking doesn't work when it relies on inputs from things like processing arrays. Building more works to an extent, but not with needing 4 per tier. 

My final air processing setup was 24 arrays and 20 freezers iirc, if it was 25s at mv and is now 120s at HV, I would have needed 384 vacuum freezers to keep up, or I could run 24 of them on ZPM to handle the 24 arrays on EV lol. Vacuum freezing as a whole just sucks, and lengthening times really doesn't seem like the way to remedy that

**Comment by NeroOneTrueKing** [comment]
@draknyte1 
Assumptions:
....
Assuming you have an infinite supply of gelid cryotheum
Assuming you are not overclocking and are instead building more machines (since that doesn't lose efficiency)
Assuming the freezing recipe does not change from current 120s per cell, as you suggest
....
Calculations:
....
The goal is to run 1 LV centrifuge full-time:
53 cells per 73.20s, or 0.724044 cells/s
or, a production time of at most 1.3811315 seconds per cell

One Factory Grade Vac Freezer performs 4 operations at once, at double speed (effectively 8x):
120s / 8x = 15 s/cell

15 / 1.3811315 = 10.8606 advanced multiblock freezers required
Thus, it would require 11 of the 'super' efficient freezers, powered at HV, to supply one LV centrifuge.
....
If it is expected that 1 or 2 advanced vacuum freezers would be able to supply enough liquid air for an LV centrifuge, we will need to modify the time on the recipe.

For a single advanced vacuum freezer to be exactly sufficient, the recipe time would have to be reduced to 11.05 seconds.

The suggested change is to reduce the recipe time to 12.00 seconds per cell, whereupon 1 or 2 advanced vacuum freezers (or 9 regular ones, as Anarak calculated) would be able to meet the demand of a single LV centrifuge.

**Comment by 0lafe** [comment]
From what I've learned about gt++ math, it's actually 4x the amount of freezers. 4 at once means it underclocks once if it can, which it can not in this case as the recipe is the same voltage as provided. So it's only a 2x bonus, which means 44, however there's no reason not to overclock here as you lose no efficiency. 11 EV ones will take the same energy and do it in the same time.

**Comment by mitchej123** [comment]
Was there some game breaking mechanic about 12 seconds of freezing oxygen before? I just don't see wtf was the point of this nerf?  12 seconds to 2 minutes and over a million EU seems a bit excessive.


Getting argon before was already pretty slow and painful.  Had a PA of 64 compressors, and then one vac freezer could keep up with that, but not the centrifuge which was always sitting around waiting.

**Comment by Prewf** [comment]
> I just don't see wtf was the point of this nerf?

The million dollar question behind the many seemingly random nerfs thrown around the last few patches.

**Comment by Dream-Master** [comment]
Ok I try to explain it. Long time ago I add compressed oxygen and compressed  nitrogen to the game via core mod(two new gases). It used 16 Oxygen/nitrogen cell to make. This cell had a vacuum freezer recipe to becomes liquid oxygen/Nitrogen. Sadly it never worked 100%. If you put a universal fluid cell in the vacuum freezer you got after 2400 ticks liquid oxygen no matter what cell you put in. No I decided to disable the two decompressed cell recipes and change the vacuum freezer recipe. Maybe 2 mins to long but 12 secs very short.

**Comment by noobyaran9** [comment]
Maybe 2 mins to long but 12 secs very short.

12s is not short at all. Its a perfect number.

**Comment by mitchej123** [comment]
Yes, please revert the liquid air change; it impacts a lot of people's setup for no good reason.

I'm fine with adding an expensive liquid oxygen/nitrogen recipe and would be open to help with that.

**Comment by Dream-Master** [comment]
ok 12 secs for liquid air

**Comment by mitchej123** [comment]
What's the expected use for liquid oxygen and liquid nitrogen?

**Comment by Dream-Master** [comment]
liquid oxygen you can use for filling tanks for your space walk on Moon or Mars etc.

**Comment by mitchej123** [comment]
ah ok, so looking to replace the convoluted recipe it had now with a simpler one?

**Comment by Dream-Master** [comment]
@NeroOneTrueKing @mitchej123  ok now ?

**Comment by mitchej123** [comment]
For comparision it was ~1.2 seconds and 3,000 EU before, this is 12 seconds and how many EU?

**Comment by Dream-Master** [comment]
i change liquid air from 2400 to 240 ticks. Change liquid oxy and nitro from 2400 to 1200 ticks

**Comment by mitchej123** [comment]
hah, the process actually looks right 


![process](https://user-images.githubusercontent.com/1894689/41501568-f44556f4-715b-11e8-85ae-5d94c1bc880c.png)

**Comment by Dream-Master** [comment]
@mitchej123 implement it if you can :P

**Comment by mitchej123** [comment]
re: liquid air - It means people are going to stack 10x more vac freezers than they did before which is terrible for TPS.

Should add in support bigger cells (more at a time).  Right now the amount of argon produced is minuscule.

**Comment by mitchej123** [comment]
that's basically how it works now, compressor --> vac freezer --> centrifuge --> nitrogen/argon/oxygen

My complaint is now the vac freezer is the huge bottleneck and people will just build a bunch more and hammer tps.  One cell taking that long doesn't make much sense; would be better to add bigger cells that net net are faster per volume but cost more

**Comment by Dream-Master** [comment]
idk how to add the bigg cells

**Comment by 0lafe** [comment]
I was always made argon at a reasonable rate, if not a little faster than I needed it. That said, it's taken too many vac freezers every time. Bigger cells would really be the best thing ever for air processing, if the recipes are done right. 4x the air done at once for 4x the power, in the same time would be an absolute godsend. Cell at LV, 4B cell at MV, 16B cell at HV, and so on. The compressing costs can increase, because that's possible to compensate for, in a way you cant with vacuum freezers.

**Comment by mitchej123** [comment]
I'd vote for completely reverting the change here, and then having an RFC where we plan out the output rates, energy, and increased volumes. Olafe's got a bunch of experience with argon and we could also call on some of the other players who hit higher tiers.

**Comment by Dream-Master** [comment]
but liquid oxy and nitro i like to change like it did it in the commit. I will not total revert it because before the recipes not working. Liquid air we can let open so longe we find a solution.

**Comment by richardhendricks** [comment]
This ticket is hard to follow. Closing ticket, open a new one if there are still concerns on liquid air/oxygen/nitrogen time
